src/js/angular/import/directives/import-resource-tree.directive.js

```diff
@@ -22,30 +30,93 @@ function importResourceTreeDirective() {
             // Public variables
             // =========================
             $scope.displayResources = [];
+            $scope.TYPE_FILTER_OPTIONS = TYPE_FILTER_OPTIONS;
+            $scope.filterByType = TYPE_FILTER_OPTIONS.ALL;
+            $scope.filterByFileName = '';
 
             // =========================
             // Private variables
             // =========================
             let selectedResources = [];
 
+            // =========================
+            // Public functions
+            // =========================
+            $scope.selectionChanged = (($event, resource) => {
+                resource.setSelection(resource.selected);
+                selectedResources = $scope.resources.getAllSelected();
+            });
+
+            $scope.changeTypeFilter = (newType) => {
```

Comment:
These three event handlers have very inconsistent names: selectionChanged, changeTypeFilter, handleInputChange and only the first one sounds like an event handler. Please, choose one convention and stick with it.